File: 0905-length-of-longest-fibonacci-subsequence/0905-length-of-longest-fibonacci-subsequence.java
class Solution {
    public int lenLongestFibSubseq(int[] arr) {
        int maxi=0;
        Set<Integer> hash=new HashSet<>();
        for(int i:arr){
            hash.add(i);
        }
        for(int i=0;i<arr.length;i++){
            for(int j=i+1;j<arr.length;j++){
                int count=2;
                int prev=arr[j];
                int curr=arr[i]+arr[j];
                while(hash.contains(curr)){
                    count++;
                    int temp=curr;
                    curr+=prev;
                    prev=temp;
                    maxi=Math.max(maxi,count);
                }
            }
        }
        return maxi;
    }
}

File: 0905-length-of-longest-fibonacci-subsequence/0905-length-of-longest-fibonacci-subsequence.py
class Solution:
    def lenLongestFibSubseq(self, arr: List[int]) -> int:
        maxi=0
        nset=set(arr)
        for i in range(len(arr)):
            for j in range(i+1,len(arr)):
                prev=arr[j]
                curr=prev+arr[i]
                count=2
                while curr in nset:
                    count+=1
                    prev,curr=curr,curr+prev
                    maxi=max(count,maxi)
        return maxi
                    

            